identity/aziot-identityd/src/main.rs

```diff
@@ -45,31 +45,43 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
 	let server = aziot_identityd::Server::new(settings, authenticator, authorizer)?;
 	let server = std::sync::Arc::new(futures_util::lock::Mutex::new(server));
 	{
-		log::info!("Provisioning starting..");
-
 		let mut server_ = server.lock().await;
-		let device = server_.provision_device().await?;
+		let device_status;
+
+		log::info!("Provisioning starting..");
+		let provisioning = server_.provision_device().await?;
 		log::info!("Provisioning complete.");
 
-		let curr_device_info_state = aziot_identityd::settings::DeviceInfo { hub_name: device.iothub_hostname, device_id: device.device_id };
+		if let aziot_identity_common::ProvisioningStatus::Provisioned(device) = provisioning {
```

Comment:
You can assign the result of a block's execution like so 
```suggestion
		let device_state = { if let aziot_identity_common::ProvisioningStatus::Provisioned(device) = provisioning {
```